hasher-matcher-actioner/hmalib/lambdas/actions/action_evaluator.py

```diff
@@ -90,46 +97,69 @@ def lambda_handler(event, context):
     return {"evaluation_completed": "true"}
 
 
-def get_action_labels(match_message: MatchMessage) -> t.List["ActionLabel"]:
+def get_action_labels(
+    match_message: MatchMessage, action_rules: t.List[ActionRule]
+) -> t.Set[ActionLabel]:
     """
-    TODO finish implementation
-    Returns an ActionLabel for each ActionRule that applies to a MatchMessage.
+    Returns action labels for each action rule that applies to a match message.
     """
-    action_rules = get_action_rules()
-    action_labels: t.List["ActionLabel"] = []
-    for action_rule in action_rules:
-        if action_rule_applies_to_match_message(
-            action_rule, match_message
-        ) and not action_labels.__contains__(action_rule.action_label):
-            action_labels.append(action_rule.action_label)
+    classifications_by_match = get_classifications_by_match(match_message)
+    action_labels: t.Set[ActionLabel] = set()
+    for classifications in classifications_by_match:
+        for action_rule in action_rules:
+            if action_rule_applies_to_classifications(action_rule, classifications):
+                action_labels.add(action_rule.action_label)
     action_labels = remove_superseded_actions(action_labels)
     return action_labels
 
 
-def get_action_rules() -> t.List["ActionRule"]:
+def get_classifications_by_match(match_message: MatchMessage) -> t.List[t.Set[Label]]:
     """
-    TODO implement
+    Creates a list of sets of classifications (as labels). Each set contains the labels that
+    classify one matching banked piece of content.
+    """
+    classifications_by_match: t.List[t.Set[Label]] = list()
+
+    for banked_signal in match_message.matching_banked_signals:
+        classifications: t.Set[Label] = set()
+        classifications.add(Label("BankSource", banked_signal.bank_source))
+        classifications.add(Label("BankId", banked_signal.bank_id))
+        classifications.add(Label("BankedContentId", banked_signal.banked_content_id))
+        for classification in banked_signal.classifications:
+            classifications.add(ClassificationLabel(classification))
+        classifications_by_match.append(classifications)
```

Comment:
blocking: What's the difference between the classification labels stored in `banked_signal.classifications` and the ones you're adding here. Why aren't the BankSource BankId, and BankedContentID labels considered classification labels.

Conceptually, I think it makes more sense to me for all the labels to be added in the Matcher before it gets to the action evaluator.

Additionally, if we do think more labeling has to be done in the actioner I think we should add it to the MatchMessage rather than returning a separate set of ClassificationLabels

Reply:
@jeberl I want to say I took this as guidance from @Dcallies. The concept of "what a match is" is determined by the matcher. Representing that in the consistent way that the action evaluator will evaluate is a concern of the evaluator. We'll have to check in with David and see where to go from here.

Reply:
There are three particular attributes, `bank_source`, `bank_id`, and `banked_signal_id` that are IDs. It seemed like a good idea to leverage `Label` to create something where the "name" of the thing was the `key` and the ID value was the `value` of a `Label`. I'm happy to create `BankSourceClassificationLabel`, `BankIDClassificationLabel` and `BankedSignalIDClassificationLabel` if that makes more sense.